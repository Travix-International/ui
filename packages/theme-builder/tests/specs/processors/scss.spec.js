const scssProcessor = require('../../../src/processors/scss.js');

describe('SCSS processor', () => {
  let doc;

  beforeEach(() => {
    doc = {
      do: {
        some: {
          string: 'aaa',
          num: '12px',
          color: '#fff'
        }
      }
    };
  });

  it('should compile doc to SCSS format', () => {
    const result = scssProcessor.compile(doc);
    expect(result).toMatchSnapshot();
  });

  it('should compile doc to SCSS format with prefix', () => {
    const result = scssProcessor.compile(doc, 'pre');
    expect(result).toMatchSnapshot();
  });
});
