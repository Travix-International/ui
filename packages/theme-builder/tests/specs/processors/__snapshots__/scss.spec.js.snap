exports[`SCSS processor should compile doc to SCSS format 1`] = `
Array [
  "$do-some-string: aaa;",
  "$do-some-num: 12px;",
  "$do-some-color: #fff;",
]
`;

exports[`SCSS processor should compile doc to SCSS format with prefix 1`] = `
Array [
  "$pre-do-some-string: aaa;",
  "$pre-do-some-num: 12px;",
  "$pre-do-some-color: #fff;",
]
`;
